src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,149 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { ReactNode, useState } from 'react';
+import { useDrag } from 'react-dnd';
+
+import { DropZone } from '@components/DropZone';
+import type { TreeItemProps } from '@components/Tree/types';
+import { useTreeContext } from '@components/Tree/TreeContext';
+import { DraggableItem, DropZonePosition } from '@utilities/dnd';
+import { merge } from '@utilities/merge';
+
+export const TreeItem = ({ id, sort, label, contentComponent, onSelect, onDrop, children }: TreeItemProps) => {
+    const { treeId, selectedIds, onSelect: onItemSelect, draggable } = useTreeContext();
+
+    const [expanded, setExpanded] = useState<boolean>(false);
+    const [hovered, setHovered] = useState<boolean>(false);
+
+    const handleMouseEnter = () => setHovered(true);
+    const handleMouseLeave = () => setHovered(false);
+
+    const draggingOpacity = 0.4;
+    const defaultOpacity = 1;
```

Comment:
Not mandatory, but to keep consistency across all files, wouldn't it be better to define these constants outside of the component scope, also avoiding recreating them on every execution, and use uppercase chars and underlines? Like `const DRAGGING_OPACITY = 0.4;`